eureka/S3_data_reduction/sigrej.py

```diff
@@ -1,63 +1,59 @@
-# $Author: carthik $
-# $Revision: 267 $
-# $Date: 2010-06-08 22:33:22 -0400 (Tue, 08 Jun 2010) $
-# $HeadURL: file:///home/esp01/svn/code/python/branches/patricio/photpipe/lib/sigrej.py $
-# $Id: sigrej.py 267 2010-06-09 02:33:22Z carthik $
-
-
 import numpy as np
 from ..lib import medstddev as msd
 
+
 def sigrej(data, sigma, mask=None,     estsig=None,   ival=False, axis=0,
-            fmean=False, fstddev=False, fmedian=False, fmedstddev=False):
-  '''This function flags outlying points in a data set using sigma rejection.
+           fmean=False, fstddev=False, fmedian=False, fmedstddev=False):
+    '''This function flags outlying points in a data set using sigma rejection.
 
     Parameters
     ----------
-    data:   ndarray
-      Array of points to apply sigma rejection to.
-    sigma:  ndarray (1D)
-      1D array of sigma values for each iteration of sigma rejection.
-      Number of elements determines number of iterations.
-    mask: (optional) byte array
-      Same shape as Data, where 1 indicates the corresponding element in
-      Data is good and 0 indicates it is bad.  Only
-      rejection of good-flagged data will be further
-      considered.  This input mask is NOT modified in the caller.
-    estsig: ndarray
-      [nsig] array of estimated standard deviations to
-      use instead of calculated ones in each iteration.
-      This is useful in the case of small datasets with
-      outliers, in which case the calculated standard
-      deviation can be large if there is an outlier and
-      small if there is not, leading to rejection of
-      good elements in a clean dataset and acceptance of
-      all elements in a dataset with one bad element.
-      Set any element of estsig to a negative value to
-      use the calculated standard deviation for that iteration.
-    ival: ndarray (2D)
-      (returned) 2D array giving the median and standard deviation (with
-      respect to the median) at each iteration.
-    axis: int
-      The axis along which to compute the mean/median.
-    fmean:  ndarray
-      (returned) the mean of the accepted data.
-    fstddev:  ndarray
-      (returned) the standard deviation of the accepted data with respect to the mean.
-    fmedian:  ndarray
-      (returned) the median of the accepted data.
-    fmedstddev: ndarray
-      (returned) the standard deviation of the accepted data with respect to the median.
+    data : ndarray
+        Array of points to apply sigma rejection to.
+    sigma : ndarray (1D)
+        1D array of sigma values for each iteration of sigma rejection.
+        Number of elements determines number of iterations.
+    mask: byte array; optional
```

Comment:
So required parameters need a space before the colon and optional input parameters don't (`data :` vs `mask:`)?

Reply:
No, sorry - this was just a result of me merging in your Astraeus PR after having edited most of the code already and missing the occasional line from your edits. I'll try to read through everything again (or even better, try to find a regex automated way of searching for things)

Reply:
Actually there were several places I'd missed, and regex came to save the day.